### PR TITLE
Fix TypeScript empty template SDK version mismatch and add E2E tests

### PR DESCRIPTION
**Pull request by davidfowl**
## Description

The `aspire new aspire-ts-empty` (Empty TypeScript AppHost) template fails with `No language support found for: typescript/nodejs` when the staging channel is configured.

### Root Cause

`ScaffoldingService.ResolveSdkVersionAsync()` queried the NuGet staging feed for `Aspire.ProjectTemplates` and picked the latest by semver. The staging feed contains stale `13.3.0-preview.1.*` packages from before the `Aspire.TypeSystem` extraction (#15028), which moved `ILanguageSupport` from `Aspire.Hosting.Ats` → `Aspire.TypeSystem`. Since `13.3.0 > 13.2.0` in semver, the resolution picked an older 13.3.0 build whose codegen DLL implements the **old** `Aspire.Hosting.Ats.ILanguageSupport` interface, while the bundled 13.2 server expects the **new** `Aspire.TypeSystem.ILanguageSupport`. The types are incompatible, so `LanguageSupportResolver` never discovers TypeScript support.

### Changes

- **Centralize SDK version** in `VersionHelper.GetDefaultSdkVersion()` — the CLI version **is** the SDK version, no need to query the NuGet feed. This matches the behavior of `GuestAppHostProject` (used by the Express/React starter) which already hardcodes the bundled version.
- **Remove feed query** from `ScaffoldingService` — eliminated `IPackagingService` and `IConfigurationService` dependencies that were only used for the now-removed feed query.
- **Add TypeScript Empty AppHost E2E test** — this template had zero E2E coverage, which is why the bug wasn't caught.
- **Add `aspire start` verification to both empty template E2E tests** — both C# and TS empty templates now verify the scaffolded project actually starts and the dashboard is reachable.
- **Extract `AspireStartAsync`/`AspireStopAsync` helpers** — reduces duplication across E2E tests and automatically dumps CLI logs on failure for debugging.

Fixes #15313

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [x] Yes
  - [ ] No
- Did you add public API?
  - [ ] Yes
  - [x] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
  - [x] No
- Does the change require an update in our Aspire docs?
  - [ ] Yes
  - [x] No

### CODE REVIEW

File: src/Aspire.Cli/Utils/VersionHelper.cs [hunk 1]
```diff
@@ -12,4 +12,28 @@ public static string GetDefaultTemplateVersion()
...
+        var plusIndex = version.IndexOf('+');
+        if (plusIndex > 0)
+        {
+            version = version[..plusIndex];
+        }
+
+        // Dev versions (e.g., "13.2.0-dev") don't exist on NuGet, fall back to latest stable
+        if (version.EndsWith("-dev", StringComparison.OrdinalIgnoreCase))
+        {
+            return "13.2.0";
```
Comment (JamesNK) [inline src/Aspire.Cli/Utils/VersionHelper.cs]: The hardcoded `"13.1.0"` fallback will be stale once 13.2.0 ships. Since this is on the `release/13.2` branch, it should probably be `"13.2.0"` already — or better, computed by stripping the prerelease suffix (e.g. take everything before `-dev` and use that).

File: src/Aspire.Cli/Scaffolding/ScaffoldingService.cs [hunk 2]
```diff
@@ -233,40 +226,10 @@ private async Task GenerateCodeViaRpcAsync(
...
-
-        if (latestPackage is null)
-        {
-            _logger.LogWarning("No packages found in channel '{Channel}', using default SDK version", channelName);
-            return DotNetBasedAppHostServerProject.DefaultSdkVersion;
-        }
-
-        _logger.LogDebug("Resolved SDK version {Version} from channel {Channel}", latestPackage.Version, channelName);
-        return latestPackage.Version;
+        return VersionHelper.GetDefaultSdkVersion();
```
Comment (JamesNK) [inline src/Aspire.Cli/Scaffolding/ScaffoldingService.cs]: `ResolveSdkVersion()` is now a single-call-site pass-through with no added logic — it was previously justified by containing the feed-query logic. Consider inlining `VersionHelper.GetDefaultSdkVersion()` directly at the call site and removing this wrapper.

File: src/Aspire.Cli/Utils/VersionHelper.cs [hunk 3]
```diff
@@ -28,10 +28,11 @@ public static string GetDefaultSdkVersion()
             version = version[..plusIndex];
         }
 
-        // Dev versions (e.g., "13.2.0-dev") don't exist on NuGet, fall back to latest stable
+        // Dev versions (e.g., "13.2.0-dev") don't exist on NuGet, strip the prerelease suffix
         if (version.EndsWith("-dev", StringComparison.OrdinalIgnoreCase))
         {
-            return "13.2.0";
+            var dashIndex = version.LastIndexOf("-dev", StringComparison.OrdinalIgnoreCase);
+            return version[..dashIndex];
```
Comment (JamesNK) [inline src/Aspire.Cli/Utils/VersionHelper.cs]: I'm not sure what the right thing to do here is. If someone has `13.3.0-dev`, should that resolve to `13.3.0`? Will that work when there isn't a stable version of 13.3.0 and it won't be available until near the end of development.

I don't know the rules here, just raising questions. Maybe 13.1.0 is right until 13.2.0 ships 🤷

**Review by JamesNK — APPROVED** [review]
I don't know what the rule should be about `-dev`. Changes look good otherwise.

File: src/Aspire.Cli/Utils/VersionHelper.cs (hunk 3 above)
Comment (davidfowl) [reply src/Aspire.Cli/Utils/VersionHelper.cs]: Good question. The `-dev` suffix only appears in local development builds (building from source without a version suffix). In that scenario:

| Build type | Version | After cleanup | NuGet exists? |
|---|---|---|---|
| CI/release | `13.2.0-preview.1.26167.1+hash` | `13.2.0-preview.1.26167.1` | ✅ |
| Stable | `13.2.0+hash` | `13.2.0` | ✅ |
| Local dev | `13.2.0-dev+hash` | `13.2.0` | ✅ (stable) |
| Local hive | `13.2.0-local.20260317.t1234+hash` | `13.2.0-local.20260317.t1234` | ✅ (local feed) |

For `-dev` → `13.2.0`: the stable `13.2.0` will exist on NuGet once shipped, but during development it won't. That said, the old hardcoded `13.1.0` had the same problem in the opposite direction — it pointed to a completely different release. The dynamic strip at least stays in the right major.minor.

In practice, local dev builds use the bundled server binary which doesn't need NuGet resolution at all — this version is used to seed `aspire.config.json` and resolve integration packages. If the version doesn't exist on NuGet, the restore will fail with a clear error rather than silently loading incompatible packages from a different branch (which was the original bug).

Happy to revert to a hardcoded value if you prefer — just wanted to avoid it going stale again.

File: tests/Shared/Hex1bAutomatorTestHelpers.cs [hunk 4]
```diff
@@ -249,9 +249,23 @@ await auto.WaitUntilAsync(
...
+                await auto.WaitUntilAsync(
+                    s => new CellPatternSearcher().Find("> Empty (TypeScript AppHost)").Search(s).Count > 0,
+                    timeout: TimeSpan.FromSeconds(5),
+                    description: "TypeScript Empty AppHost template selected");
+                await auto.EnterAsync();
+                break;
+
+            default:
+                throw new ArgumentOutOfRangeException(nameof(template), template, $"Unsupported template: {template}");
+        }
```
Comment (IEvangelist) [inline tests/Shared/Hex1bAutomatorTestHelpers.cs:245]: This is already addressed in #15308

Comment (davidfowl) [reply tests/Shared/Hex1bAutomatorTestHelpers.cs:245]: What do you want to do there? Merge that PR and let me rebase? It's missing the default fallthrough check.

Comment (IEvangelist) [reply tests/Shared/Hex1bAutomatorTestHelpers.cs:245]: I'm happy to close mine, if yours covers everything.

File: src/Aspire.Cli/Utils/VersionHelper.cs (hunk 3 above)
Comment (eerhardt) [reply src/Aspire.Cli/Utils/VersionHelper.cs]: > Maybe 13.1.0 is right until 13.2.0 ships

I agree with @JamesNK here. `13.2.0` hasn't shipped yet, so returning that stable, but non-existant version doesn't really make sense.

What fails if we return `13.2.0-dev` from this method?

File: tests/Shared/Hex1bAutomatorTestHelpers.cs (hunk 4 above)
Comment (davidfowl) [reply tests/Shared/Hex1bAutomatorTestHelpers.cs:245]: Yep, lets do that.

**Review by IEvangelist — APPROVED** [review]

File: src/Aspire.Cli/Utils/VersionHelper.cs (hunk 3 above)
Comment (davidfowl) [reply src/Aspire.Cli/Utils/VersionHelper.cs]: Good point. If we return `13.2.0-dev`, the downstream code tries to restore `Aspire.Hosting` version `13.2.0-dev` from NuGet — which doesn't exist. Same problem with `13.2.0` during development (hasn't shipped).

But `13.2.0-dev` only appears in local source builds without a version suffix (not CI, not localhive). In that case, the NuGet restore will fail regardless because there's no matching package on any feed. The question is whether we want a clear NuGet error ("package not found") or a confusing version mismatch error (the original bug).

I think the cleanest option is to just pass through `13.2.0-dev` as-is and let the restore fail with an obvious error. Should I revert the `-dev` stripping entirely?

Comment (davidfowl) [reply src/Aspire.Cli/Utils/VersionHelper.cs]: Removed the `-dev` special case entirely. If you build everything locally with `./build.sh`, the packages are also `13.2.0-dev` — so passing it through is correct. It will resolve from the local package artifacts (if configured as a NuGet feed). No reason to mangle it.
